### PR TITLE
Feature/aqi 3025

### PR DESCRIPTION
**Pull request by AQI-DanG**

### CODE REVIEW

**Review by joezhou-aqi — APPROVED** [review]

**Review by ChrisMarin-AI — APPROVED** [review]

File: proto/one/common/configuration/configuration_note.proto [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    string configurationId = 2;
+
+    // time associated with the note
+    JsonTicksDateTime noteTime = 3;
+
+    // text of the note
+    string note = 4;
+    
+    // tags associated with this note
+    repeated string tags = 5;
```
Comment (JonathanRobertson-AQI) [inline proto/one/common/configuration/configuration_note.proto:22]: I think we need a separate file for tags to accomplish https://aquaticinformatics.atlassian.net/wiki/spaces/OP/pages/59080751/Method+Claros.Common.Configuration.API+Notes+Tags+GET

**Review by JonathanRobertson-AQI — CHANGES_REQUESTED** [review]
please create a new file for ConfigurationTag to match its table structure and have ConfigurationNote contain a collection ConfigurationTags, instead of strings

File: proto/one/common/configuration/configuration_note.proto (hunk 1 above)
Comment (AQI-DanG) [reply proto/one/common/configuration/configuration_note.proto:22]: Was planning to implement that endpoint using the existing KeyValue protobuf model where the key would be the configurationId and the value would be the space delimited list of tags.

**Comment by AQI-DanG** [comment]
> please create a new file for ConfigurationTag to match its table structure and have ConfigurationNote contain a collection ConfigurationTags, instead of strings

The only other fields on the ConfigurationTag table are the ConfigurationId and the NoteId both of which are already available on the ConfigurationNote object, I do not see any value in duplicating that information.

**Comment by JonathanRobertson-AQI** [comment]
> > please create a new file for ConfigurationTag to match its table structure and have ConfigurationNote contain a collection ConfigurationTags, instead of strings
> 
> The only other fields on the ConfigurationTag table are the ConfigurationId and the NoteId both of which are already available on the ConfigurationNote object, I do not see any value in duplicating that information.

I think that at a minimum, we need the Created audit fields on Tags.  I'd lean towards using the standard audit fields that we have on other tables.  I'm chewing on the value of having full CRUD on Tags, like Reports.  Ultimately, I don't really think we need it. but I wonder if we should at least have a Delete endpoint for cleanup.

**Review by AQI-DanG — COMMENTED** [review]
Looks good to me but I cannot approve because I created the PR

**Review by joezhou-aqi — APPROVED** [review]

**Review by JonathanRobertson-AQI — APPROVED** [review]

**Pull request merged**